src/Const/Commands.ts

```diff
@@ -0,0 +1,237 @@
+        for (let i = 0; i < AIs.length; ++i) {
+            if (!(AIs[i].owner instanceof TEntity)) continue;
+            client.possess(AIs[i]);
+        }
+    },
+    admin_godmode: (client: Client) => {
+        if(client.camera?.camera.player?.style?.styleFlags) {
+            if(client.camera.camera.player.style.styleFlags & StyleFlags.invincibility) {
+                client.camera.camera.player.style.styleFlags ^= StyleFlags.invincibility;
+            } else {
+                client.camera.camera.player.style.styleFlags |= StyleFlags.invincibility;
+            }
+        }
+    },
+    admin_summon: (client: Client, entityArg: string, countArg?: string, xArg?: string, yArg?: string) => {
+        const count = countArg ? parseInt(countArg) : 1;
+        const x = parseInt(xArg || "");
+        const y = parseInt(yArg || "");
+        const game = client.camera?.game;
+        const TEntity = {
+            "Defender": Defender,
+            "Summoner": Summoner,
+            "Guardian": Guardian,
+            "FallenOverlord": FallenOverlord,
+            "FallenBooster": FallenBooster,
+            "FallenAC": FallenAC,
+            "FallenSpike": FallenSpike,
+            "ArenaCloser": ArenaCloser,
+            "Crasher": Crasher,
+            "Pentagon": Pentagon,
+            "Square": Square,
+            "Triangle": Triangle,
+        }[entityArg];
+
+        if (isNaN(count) || count < 0 || !game || !TEntity) return;
+
+        for (let i = 0; i < count; ++i) {
+            const boss = new TEntity(game);
+            if (!isNaN(x) && !isNaN(y)) {
+                boss.position.x = x;
+                boss.position.y = y;
+            }
+        }
+    },
+    admin_kill_all: (client: Client) => {
+        const game = client.camera?.game;
+        if(!game) return;
+        for (let id = 0; id <= game.entities.lastId; ++id) {
+			const entity = game.entities.inner[id];
+			if (Entity.exists(entity) && entity instanceof LivingEntity && entity !== client.camera?.camera.player) entity.health.health = 0;
+		}
+    },
+    admin_close_arena: (client: Client) => {
+        client?.camera?.game.arena.close();
+    },
+    admin_kill_entity: (client: Client, entityArg: string) => {
+        const TEntity = {
+            "ArenaCloser": ArenaCloser,
+            "Dominator": Dominator,
+            "Bullet": Bullet,
+            "Tank": TankBody,
+            "Shape": AbstractShape,
+            "Boss": AbstractBoss
+        }[entityArg];
+        const game = client.camera?.game;
+        if (!TEntity || !game) return;
+
+        for (let id = 0; id <= game.entities.lastId; ++id) {
+			const entity = game.entities.inner[id];
+			if (Entity.exists(entity) && entity instanceof TEntity) entity.health.health = 0;
+		}
+    }
+} as Record<CommandID, CommandCallback>
+
+export const executeCommand = (client: Client, cmd: string, args: string[]) => {
+    if (!commandDefinitions[cmd as CommandID] || !commandCallbacks[cmd as CommandID]) {
```

Comment:
For `commandDefinitions` I would suggest using a `Map` to prevent prototype pollution. Alternatively, you could technically use `commandDefinitions.hasOwnProperty` instead, but still I suggest using `Map` instead of an object for the map.

Reply:
we cant easily JSON.stringify a map, which would be necessary for the /commands endpoint

Reply:
Ah makes sense. Maybe then an array of commands similar to client/config? Idk

Reply:
why would you use an array in that case? objects have faster lookup times, otherwise we would need to search the array every execution

Reply:
imo hasOwnProperty is the only option here :v

Reply:
Sounds good